virttest/qemu_devices/qdevices.py

```diff
@@ -2180,6 +2180,141 @@ def __eq__(self, other):
         return False
 
 
+class QNetDevice(QCustomDevice):
+    def __init__(self, nettype, params=None, legacy=False):
+        """
+        netdev device
+
+        :param nettype: The type of the netdev device.
+        :type nettype: str
+        :param params: The parameters of the netdev device.
+        :type params: dict
+        :param legacy: Whether the netdev device is in legacy mode (-net).
+        :type legacy: bool
+        """
+        super().__init__('netdev', params)
+        if legacy:
+            self.type = 'net'
+        self.set_param('type', nettype)
+
+    def _cmdline_raw(self):
+        """ :return: cmdline command to define this device in raw format"""
+        params = self.params.copy()
+
+        out = f'-{self.type} {params.pop("type")}'
+        for key, value in params.items():
+            if value != "NO_EQUAL_STRING":
+                if (key in ['dnssearch', 'hostfwd', 'guestfwd'] and
+                        isinstance(value, list)):
+                    for v in value:
+                        out += ",%s=%s" % (key, v)
+                else:
+                    out += ",%s=%s" % (key, value)
+            else:
+                out += ",%s" % key
+        return out
+
+    def _cmdline_json(self):
+        # Legacy mode '-net' does not support JSON format
+        if self.type == 'net':
+            return self._cmdline_raw()
+        out = "-%s " % self.type
+        params = self.params.copy()
+        params = self._convert_netdev_args(params)
+        return out + f" '{json.dumps(params)}'"
```

Comment:
```suggestion
    def _cmdline_json(self):
        out = "-%s " % self.type
        params = self.params.copy()
        params = self._convert_netdev_args(params)
        return out + f" '{json.dumps(params)}'"
```

Hello @PaulYuuu , IMO, the _cmdline_json() has only task to generate the parameter in json format.
I prefer to remove the line#2218  ~ line#2220.
Meanwhile, move the line#2218  ~ line#2220 to `cmdline()`, here's overridden `cmdline()` in `QNetdevice`:
```
    def cmdline(self):
        if self.type == "net":
            return self._cmdline_json()
        else:
            return super().cmdline()
```

Reply:
OK, that's makes sense.